### PR TITLE
Add PWR075: Avoid using GNU Fortran extensions

### PR DESCRIPTION
**Pull request by alvrogd**
Add entry and benchmark for a new Fortran-specific check. My results:

```txt
$ lscpu
  Model name:             13th Gen Intel(R) Core(TM) i7-13700H

$ gfortran --version
GNU Fortran (Debian 12.2.0-14) 12.2.0

$ ./run-benchmarks.py --check PWR075
------------------------------------------------------------------
Benchmark                        Time             CPU   Iterations
------------------------------------------------------------------
PWR075 Fortran Example        3137 us         3137 us          226
PWR075 Fortran Improved       1725 us         1724 us          404
```

### CODE REVIEW

**Comment by jgonzac** [comment]
```
$ lscpu
  Model name:             AMD Ryzen 7 4800H with Radeon Graphics

$ gfortran --version
GNU Fortran (Ubuntu 13.2.0-23ubuntu4) 13.2.0
```
```
------------------------------------------------------------------
Benchmark                        Time             CPU   Iterations
------------------------------------------------------------------
PWR075 Fortran Example        5461 us         5461 us          129
PWR075 Fortran Improved       1251 us         1251 us          564
```

**Review by jgonzac — APPROVED** [review]

**Review by d-alonso — COMMENTED** [review]
I think this is missing an update to [the index](https://github.com/codee-com/open-catalog/blob/main/README.md#checks), right?

**Comment by alvrogd** [comment]
> I think this is missing an update to [the index](https://github.com/codee-com/open-catalog/blob/main/README.md#checks), right?

Indeed, thanks for the catch! Should be [done](https://github.com/codee-com/open-catalog/compare/ecb1508e03ee063647cb89bbd330f87067c53624..6ece8ac3e6b5dc4f24610dae07dde4d403d56fd5).

**Comment by igarcia0** [comment]
> > I think this is missing an update to [the index](https://github.com/codee-com/open-catalog/blob/main/README.md#checks), right?
> 
> Indeed, thanks for the catch! Should be [done](https://github.com/codee-com/open-catalog/compare/ecb1508e03ee063647cb89bbd330f87067c53624..6ece8ac3e6b5dc4f24610dae07dde4d403d56fd5).

Wild idea: how hard would it be to write a small validator that checks that every directory `Checks/PWR***` has a link to it in `README.md`?

File: Checks/PWR075/benchmark/solution.f90 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  implicit none
+
+  real(kind=c_double), parameter :: pi = 3.141592653589793_c_double
+
+contains
+
+  real(kind=c_double) function dcosd(degrees)
+    real(kind=c_double), intent(in) :: degrees
+    ! `dcos` expects the angle to be in radians, not degrees
+    dcosd = dcos(degrees * pi / 180.0)
```
Comment (igarcia0) [inline Checks/PWR075/benchmark/solution.f90:23]: https://tauday.com/tau-manifesto
Couldn't resist.
TLDR: $\tau=circumference/radius=2\pi\approx6.28$ is a constant that makes more sense IMO.
$degrees / 360 = turns$
$turns * \tau = radians$

Feel free to ignore if confusing to those unfamiliar.

**Review by igarcia0 — APPROVED** [review]
LGTM. Kudos for beating the extensions.

File: Checks/PWR075/benchmark/solution.f90 (hunk 1 above)
Comment (alvrogd) [reply Checks/PWR075/benchmark/solution.f90:23]: Wow, it's been years since I last heard from tau :grin: 

> Feel free to ignore if confusing to those unfamiliar.

Indeed, I'd prefer to keep pi for familiarity to most readers.

**Comment by alvrogd** [comment]
The latest `flang` builds added support for `rand()`. Accordingly, [I've replaced](https://github.com/codee-com/open-catalog/compare/6ece8ac3e6b5dc4f24610dae07dde4d403d56fd5..ba84055aade53a083c6b2ebf68b023776a024081) the code example in the `README.md` (not the benchmark) with the `access()` GNU Fortran extension, which is not yet supported by `flang`.
